### PR TITLE
Update fetch to latest https://github.com/github/fetch

### PR DESCRIPTION
**Pull request by l-urence**
This fixes https://github.com/facebook/react-native/issues/6326 with updating https://github.com/facebook/react-native/blob/master/Libraries/Fetch/fetch.js to latest from  https://github.com/github/fetch. 

cc @skevy @steveluscher

### CODE REVIEW

**Comment by l-urence** [comment]
Right now this is causing hight cpu load event if app is idle...

**Comment by skevy** [comment]
@l-urence interesting. @davidaurelio apparently fixed the regression that caused #6326 in the first place...should be landing in master soon. Let's take a look at that, before we do any more work on updating `fetch`...it may not be necessary.

**Comment by davidaurelio** [comment]
Just landed: fd3a0ba781082a567d5713524fb269877baa0b1d

**Comment by skevy** [comment]
Thanks @davidaurelio

**Comment by skevy** [comment]
@l-urence if you can determine what's causing the high cpu load and fix it...we can probably land this PR (seeing as the latest version of fetch does have some bug fixes).

Otherwise, we're probably OK. Let me know how you'd like to proceed. :)

**Comment by davidaurelio** [comment]
Upgrading the library can’t hurt. But apparently we’re using a customized version. But `self` is just an object in our file, not the global object. This might have weird side effects or introduce subtle breakages.

**Comment by l-urence** [comment]
I created a new RN (https://github.com/facebook/react-native/commit/fd3a0ba781082a567d5713524fb269877baa0b1d) project with almost the same code from https://github.com/facebook/react-native/issues/6326 without updating  fetch.js. I noticed even then a hight cpu load and app is just displaying the fetched text. 
 
screen shot 2016-03-07 at 23 07 09

Maybe its just my machine... However thanks to @davidaurelio https://github.com/facebook/react-native/issues/6326 is fixed.

@davidaurelio I think the updated fetch.js uses `self` the same way as before.

**Comment by l-urence** [comment]
@skevy pulled latest from RN master into update-fetch branch. Now cpu utilisation seems to be normal if app is idle.

File: Libraries/Fetch/fetch.js [hunk 1]
```diff
@@ -151,7 +152,7 @@ var self = {};
   }
 
   var support = {
-    blob: typeof FileReader === 'function' && typeof Blob === 'function' && (function() {
+    blob: 'FileReader' in self && 'Blob' in self && (function() {
```
Comment (davidaurelio) [inline Libraries/Fetch/fetch.js]: @l-urence here

File: Libraries/Fetch/fetch.js [hunk 2]
```diff
@@ -159,7 +160,8 @@ var self = {};
         return false
       }
     })(),
-    formData: typeof FormData === 'function'
+    formData: 'FormData' in self,
+    arrayBuffer: 'ArrayBuffer' in self
```
Comment (davidaurelio) [inline Libraries/Fetch/fetch.js]: @l-urence here

**Comment by l-urence** [comment]
@davidaurelio now its checking for ArrayBuffer, Blob, FileReader, FormData in global namespace.

**Comment by davidaurelio** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**